core/src/main/java/io/grpc/internal/CachedSubchannelPool.java

```diff
@@ -30,14 +29,16 @@
 import io.grpc.LoadBalancer.SubchannelStateListener;
 import io.grpc.SynchronizationContext.ScheduledHandle;
 import java.util.HashMap;
+import java.util.List;
 import java.util.concurrent.TimeUnit;
 
 /**
  * A {@link SubchannelPool} that keeps returned {@link Subchannel}s for a given time before it's
  * shut down by the pool.
  */
-final class CachedSubchannelPool implements SubchannelPool {
-  private final HashMap<EquivalentAddressGroup, CacheEntry> cache =
+// TODO: make it package private. Right now it has to be public for grpclb testing.
+public final class CachedSubchannelPool implements SubchannelPool {
```

Comment:
I would prefer keep `CachedSubchannelPool` in place and copy the implementation in `SubchannelPoolHelper`. There is more needed to implement in the Helper, like `updateSubchannelAddresses()`, so that it's a semantic-complete `Helper` implementation.